src/cases/resolvers/activity.resolver.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { Resolve } from '@angular/router';
+import { ActivityService } from '@hmcts/ccd-case-ui-toolkit';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class ActivityResolver implements Resolve<boolean> {
+
+  constructor(private readonly activityService: ActivityService) {}
+
+  public resolve(): Observable<boolean> {
+    this.activityService.verifyUserIsAuthorized();
+    return Observable.of(true);
```

Comment:
Why does this always return true? Is it not based on whether or not the user is authorised? If not, what is the point of it? 🤷‍♂️

Reply:
I think this another instance of stuff being copied from CCD-UI. `this.activityService.verifyUserIsAuthorized();` where the actual check happens. I think you still would want the resolver to return true regardless of whether the activity call is happening or not.

Reply:
Why would you still want it to return true even if the user is not authorised? Doesn't that make the call redundant? Or perhaps it's a poorly named method...?

Reply:
Oh, `this.activityService.verifyUserIsAuthorized()` doesn't return anything. How odd. 🤯

Reply:
If the resolver returns false the navigation would not happen at all, which is not what we want. We only want the activity call to be made if it is authorised to do so. I agree that maybe a resolver is not the best way to do it.